### PR TITLE
*: back quote and escape MySQL identifier #835

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
re-implement #835 , please check if changes are proper.

### What is changed and how it works?
back quote and escape MySQL identifier

### Check List 

Tests 

 - Integration test

Code changes

Side effects

Related changes

### CODE REVIEW

**Comment by lance6716** [comment]
mydumper still can't compile but I made it 😭, PTAL @csuzhangxc

File: syncer/checkpoint.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ package syncer
 
 import (
 	"fmt"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
```
Comment (csuzhangxc) [inline syncer/checkpoint.go:18]: re-order it.

File: syncer/checkpoint.go [hunk 2]
```diff
@@ -480,15 +481,14 @@ func (cp *RemoteCheckPoint) prepare(tctx *tcontext.Context) error {
 }
 
 func (cp *RemoteCheckPoint) createSchema(tctx *tcontext.Context) error {
-	sql2 := fmt.Sprintf("CREATE SCHEMA IF NOT EXISTS `%s`", cp.schema)
+	sql2 := fmt.Sprintf("CREATE SCHEMA IF NOT EXISTS %s", dbutil.ColumnName(cp.schema))
```
Comment (csuzhangxc) [inline syncer/checkpoint.go:484]: How about quoting it when init it?

Comment (lance6716) [reply syncer/checkpoint.go]: https://github.com/pingcap/dm/pull/840/commits/54db3d8b21ab19943448e5a1de07ced2a533f56b

File: syncer/checkpoint.go (hunk 1 above)
Comment (lance6716) [reply syncer/checkpoint.go]: https://github.com/pingcap/dm/pull/840/commits/54db3d8b21ab19943448e5a1de07ced2a533f56b

File: tests/all_mode/run.sh [hunk 3]
```diff
@@ -6,6 +6,7 @@ cur=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 source $cur/../_utils/test_prepare
 WORK_DIR=$TEST_DIR/$TEST_NAME
 API_VERSION="v1alpha1"
+ILLEGAL_CHAR_NAME='t-Ë!s`t'
```
Comment (GMHDBJD) [inline tests/all_mode/run.sh:9]: Seems it is a legal name

Comment (lance6716) [reply tests/all_mode/run.sh:9]: yes, want to express ' not a valid name if not quoted in mysql' 🤣

**Pull request merged**